modules/src/ics02_client/events.rs

```diff
@@ -0,0 +1,108 @@
+use crate::attribute;
+use crate::events::{IBCEvent, TryObject};
+use crate::ics02_client::client_type::ClientType;
+use crate::ics24_host::identifier::ClientId;
+
+use serde_derive::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use tendermint::block;
+use tendermint::rpc::event_listener::ResultEvent;
+
+// TODO - find a better place for NewBlock
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct NewBlock {
+    pub height: block::Height,
+}
+
+impl NewBlock {
+    pub fn new(h: block::Height) -> NewBlock {
+        NewBlock { height: h }
+    }
+}
+
+impl TryFrom<&ResultEvent> for NewBlock {
+    type Error = Box<dyn std::error::Error>;
```

Comment:
Putting this here, but applies in general: since we are using `Box<dyn std::error::Error>` in a bunch of places we might want to introduce a type alias for it, eg. `BoxError`, or just use the one defined in [`anomaly::BoxError`](https://docs.rs/anomaly/0.2.0/anomaly/type.BoxError.html) (though the latter is a bit more restrictive as it enforces thread-safety by having `Send + Sync + 'static` bounds as well.